### PR TITLE
Init categorization script

### PR DESCRIPTION
**Pull request by dominikfranzen**
Added a script to parse minifigures.txt and generate a json-file

### CODE REVIEW

File: clustering/categories.json [hunk 1]
```diff
@@ -0,0 +1 @@
...0060', 'mar0061', 'mar0062', 'mar0063', 'mar0064', 'mar0065', 'mar0066', 'mar0067', 'mar0068', 'mar0069', 'mar0070', 'mar0071', 'mar0072', 'mar0073', 'mar0074', 'mar0075', 'mar0076', 'mar0077', 'mar0078', 'mar0079', 'mar0080', 'mar0081', 'mar0082', 'mar0083', 'mar0084', 'mar0085', 'mar0086', 'mar0087', 'mar0088', 'mar0089', 'mar0090', 'mar0091', 'mar0092', 'mar0093', 'mar0094', 'mar0096', 'mar0097', 'mar0098', 'mar0099', 'mar0100', 'mar0102', 'mar0103', 'mar0101']}{'category_id': 1125,'minifigure_ids': ['vid001', 'vid003', 'vid004', 'vid005', 'vid015', 'vid006', 'vid007', 'vid013', 'vid012', 'vid017', 'vid009', 'vid008', 'vid011', 'vid010', 'vid018', 'vid016', 'vid014', 'vid002', 'vid004a', 'vid019', 'vid020', 'vid021', 'vid022', 'vid023', 'vid024', 'vid025', 'vid026', 'vid027', 'vid028', 'vid029', 'vid030', 'vid031', 'vid032', 'vid033', 'vid034', 'vid038', 'vid042', 'vid044', 'vid037', 'vid035', 'vid040', 'vid046', 'vid045', 'vid036', 'vid043', 'vid039', 'vid041']}{'category_id': 1154,'minifigure_ids': ['ftv001', 'ftv002', 'ftv003', 'ftv004', 'ftv005', 'ftv007', 'ftv006']}{'category_id': 1163,'minifigure_ids': ['que004', 'que005', 'que006', 'que007', 'que001', 'que002', 'que003']}
```
Comment (fabianschmauder) [inline clustering/categories.json:1]: `'vid041']}{'catego`
This is no valid json.
I would not upload this file to git when you can generate it

File: clustering/cluster_minifigures.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        with open('minifigures.txt') as textfile:
+            for line in textfile:
+                if 'Category ID' in line:
+                    continue
+                content = line.strip().split('\t')
+                if len(content) < 2:
+                    continue
+                if content[0] != element:
+                    continue
+                minifigures_list.append(content[2])
```
Comment (fabianschmauder) [inline clustering/cluster_minifigures.py:29]: this looks like duplication. 
you can add a second parameter to list_categories that indicates the column and us this here

like  
`def list_column_content(file, column_index)`

File: clustering/cluster_minifigures.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            category_id = content[0]
+            if category_id not in categories_list:
+                categories_list.append(category_id)
+    return categories_list
+
+categories = list_categories('minifigures.txt')
+
+with open('categories.json', 'w') as json_file:
+    for element in categories:
+        minifigures_list = []
```
Comment (fabianschmauder) [inline clustering/cluster_minifigures.py:19]: move this and the line `json_file.write('{\'category_id\': ' + str(element) + ',\'minifigure_ids\': ' + str(minifigures_list) + '}') `outside of the for loop to create a valid json

File: clustering/cluster_minifigures.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            for line in textfile:
+                if 'Category ID' in line:
+                    continue
+                content = line.strip().split('\t')
+                if len(content) < 2:
+                    continue
+                if content[0] != element:
+                    continue
+                minifigures_list.append(content[2])
+        json_file.write('{\'category_id\': ' + str(element) + ',\'minifigure_ids\': ' + str(minifigures_list) + '}')
```
Comment (fabianschmauder) [inline clustering/cluster_minifigures.py:30]: you can also use a dict with category_id as key then you do not need to use escaping` \'`

File: clustering/cluster_minifigures.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            if len(content) < 2:
+                continue
+            category_id = content[0]
+            if category_id not in categories_list:
+                categories_list.append(category_id)
+    return categories_list
+
+categories = list_categories('minifigures.txt')
+
+with open('categories.json', 'w') as json_file:
```
Comment (fabianschmauder) [inline clustering/cluster_minifigures.py:17]: i would move write to file move to an own method like

`def write_to_json(data, file):
        with open(file, 'w') as json_file:
              json_file.write(json.dumps(data))
`

**Review by fabianschmauder — COMMENTED** [review]
Nice first PR that you share. I added some comments 😉

File: clustering/cluster_minifigures.py (hunk 3 above)
Comment (dominikfranzen) [reply clustering/cluster_minifigures.py:19]: If it was outside of the loop, wouldn't it just return the last category_id and it's minifigure id's?

File: clustering/cluster_minifigures.py (hunk 2 above)
Comment (dominikfranzen) [reply clustering/cluster_minifigures.py:29]: totally right, I'm gonna solve it :)

File: clustering/cluster_minifigures.py (hunk 3 above)
Comment (dominikfranzen) [reply clustering/cluster_minifigures.py:19]: Nevermind my comment. I managed to do as you supposed by using a list for all dict entries and it works! Big thx 
for your feedback

**Pull request merged**